data-serving/data-service/src/controllers/case.ts

```diff
@@ -86,7 +91,15 @@ export const create = async (req: Request, res: Response): Promise<void> => {
             await c.validate();
             result = c;
         } else {
-            result = await c.save();
+            if (numCases === 1) {
+                result = await c.save();
+            } else {
+                const cases = [];
+                for (let i = 0; i < numCases; i++) {
```

Comment:
Ah, I had tried `Array(numCases).fill(new Case(...))` but mongo's insertMany didn't want you to insert the same reference multiple times. This works since it creates a new Case reference for each entry. Done.